### DISCUSSION TITLE
How can I integrate diagrams/drawio with my internal GitHub Enterprise?

### DISCUSSION THREAD

**Discussion by jkilgrow** [jgraph/drawio#2355] [Q&A]
I would like to use the GitHub integration of diagrams with my corporate GitHub Enterprise server but I'm having a difficult time figuring out exactly how to do this. I keep getting stuck on the OAuth integration. Is there an OAuth integration guide or am I doing something else wrong? A little help, please?

**Comment by davidjgraph** [comment]
Maybe you could document what you did to kick off the documentation and we can correct it?

**Reply by jkilgrow** [reply]
Well, I don't know that it needs a correction, let me try to explain what I'm trying to do...

In the application is an integration with github.com. I would like to be able to edit files on github.mydomain.com (my internal github repositories). That's the whole point.

So, to that point, what I have tried so far:

* Downloaded the jgraph/drawio code and uploaded it into a new repo in github.mydomain.com, published master to gh pages
* Downloaded the jgraph/docker-drawio code and uploaded it into a new repo in github.mydomain.com
    * I have tried a number of things with the standalone docker image. The thing that I have gotten closest with is `docker cp newPreConfig.js` into the running docker image to get it to hit github.mydomain.com.
 
But, all of this leads me back to one thing and that is I don't think I know enough about the drawio application to be able to configure it as an oauth application in github.mydomain.com. I say this because I seem to hit the same oauth wall. I can't even discern _what_ the actual issue is because I don't know where to look for error logs, etc. Hence, my call for help. 

So, I don't know that the drawio/diagrams.net app is wrong (it's really lovely!), it's my cruddy knowledge of things. Any help would be great!

**Comment by DouglasLino** [comment]
Hi, were you able to solve it? It's been a couple of years now, and the current web version does support GitHub integration and works perfectly!
I'm trying to do the same thing locally, but I can't find a way to get draw.io running locally and connect it to github.com.

[a comment by an automated account is not shown]
